### PR TITLE
[OMSICpp/cppRuntime] Extended cpp runtime to OMSI Cpp library

### PR DESCRIPTION
**Pull request by niklwors**
Renamed cpp runtime to omsicpp runtime
omsicpp code target is later used to extend omsic codegeneration with additonal
functionality
OMSICpp Simulation runtime for FMU 2.0 ME and OMSUC simulation in C++.
 - Added OMSICpp build
   - Use CMake to build SimultaionRuntime/OMSICpp
 - Added OMSICpp folder in SimulationRuntime
   - The code targets cpp and cpp fmu using this runtime also now
   - Using OMSIBase library for base functionalities shared with OMSIC runtime
   - Wrapper class for FMI 2.0 ModelExchange functions
     - Functions for continuous simulation of FMU/OMSU
     - Functions for event simulation of FMU/OMSU
     - Getter and Setter functions for FMU/OMSU
     - Logging and some debugging functionalities
     - Initialization and deallocation of FMU/OMSU
   - OMSICpp Simulation runtime which supports omsu/fmu 2.0 ME and  the old
     cpp code traget systems
 - Documentation with Doxygen
   - Doxyfile not included

Co-authored-by: AnHeuermann <andreas.heuermann@fh-bielefeld.de>
Co-authored-by: wibraun <wbraun@fh-bielefeld.de>

### CODE REVIEW

**Comment by lochel** [comment]
@niklwors There are conflicts which need to be resolved before the changes can be merged.

File: OMCompiler/Compiler/SimCode/SimCodeMain.mo [hunk 1]
```diff
@@ -781,7 +835,10 @@ algorithm
         runTplWriteFile(func = function CodegenOMSIC.generateOMSIC(a_simCode=simCode), file=simCode.fullPathPrefix+"/"+fileprefix+"_omsic.c");
 
         runTpl(func = function CodegenOMSI_common.generateEquationsCode(a_simCode=simCode, a_FileNamePrefix=fileprefix));
+		
+		   runTpl(func = function CodegenOMSICpp.translateModel(a_simCode=simCode, a_FMUVersion=FMUVersion, a_FMUType=FMUType));
```
Comment (adrpo) [inline OMCompiler/Compiler/SimCode/SimCodeMain.mo]: Please remove tabs.

**Comment by adrpo** [comment]
Seems somehow that the build omc has issues with parsing?!

```
(gdb) bt
#0  0x00007ffff4c3ceba in omc__escapedStringLength (str=0x5 <error: Cannot access memory at address 0x5>, nl=0, hasEscape=0x7fffffffa790) at ./util/modelica_string.c:258
#1  0x00007ffff4c3d025 in omc__escapedString (str=0x5 <error: Cannot access memory at address 0x5>, nl=0) at ./util/modelica_string.c:286
#2  0x00007ffff4f016cb in System_escapedString (str=0x5 <error: Cannot access memory at address 0x5>, nl=0) at System_omc.c:684
#3  0x00007ffff6b5a32d in omc_System_escapedString (threadData=0x7fffffffe848, _unescapedString=0x0, _unescapeNewline=0 '\000') at /var/lib/jenkins/ws/OpenModelica_PR-690/OMCompiler/Compiler/Util/System.mo:841
#4  0x00007ffff70494c9 in omc_ValuesUtil_valString2 (threadData=0x7fffffffe848, _inValue=0x7fffd7aa9643) at /var/lib/jenkins/ws/OpenModelica_PR-690/OMCompiler/Compiler/FrontEnd/ValuesUtil.mo:1758
#5  0x00007ffff704abb4 in omc_ValuesUtil_valListString (threadData=0x7fffffffe848, _inValueLst=0x7fffd7aa9623) at /var/lib/jenkins/ws/OpenModelica_PR-690/OMCompiler/Compiler/FrontEnd/ValuesUtil.mo:2005
#6  0x00007ffff7049742 in omc_ValuesUtil_valString2 (threadData=0x7fffffffe848, _inValue=0x7fffd7a9b903) at /var/lib/jenkins/ws/OpenModelica_PR-690/OMCompiler/Compiler/FrontEnd/ValuesUtil.mo:1782
#7  0x00007ffff7048a2b in omc_ValuesUtil_valString (threadData=0x7fffffffe848, _inValue=0x7fffd7a9b903) at /var/lib/jenkins/ws/OpenModelica_PR-690/OMCompiler/Compiler/FrontEnd/ValuesUtil.mo:1722
#8  0x00007ffff6faa84b in omc_Interactive_evaluateAlgStmt (threadData=0x7fffffffe848, _inAlgorithmItem=0x7fffd4b57c33) at /var/lib/jenkins/ws/OpenModelica_PR-690/OMCompiler/Compiler/Script/Interactive.mo:464
#9  0x00007ffff6facba1 in omc_Interactive_evaluate2 (threadData=0x7fffffffe848, _inStatements=0x7fffd4c153c3) at /var/lib/jenkins/ws/OpenModelica_PR-690/OMCompiler/Compiler/Script/Interactive.mo:349

```
Printing _value in frame 8 returns string=0 which shouldn't be!

```
(gdb) call anyString(_value)
$2 = 0x737590 "Values.Value.ARRAY(valueLst = {Values.Value.STRING(string = FmuExportCrossCompile), Values.Value.STRING(string = 0)}, dimLst = {2})"
```

This is why the cross-build-fmu stage is failing.

**Comment by adrpo** [comment]
As far as I can see it dies while evaluating this for loop:
https://github.com/OpenModelica/OpenModelica/blob/master/testsuite/special/FmuExportCrossCompile/fmuExportCrossCompile.mos#L62

**Comment by adrpo** [comment]
@niklwors, can you try to run testsuite/special/FmuExportCrossCompile/fmuExportCrossCompile.mos with your omc compiler so we can see if you can reproduce this?

**Comment by niklwors** [comment]
> 
> 
> @niklwors, can you try to run testsuite/special/FmuExportCrossCompile/fmuExportCrossCompile.mos with your omc compiler so we can see if you can reproduce this?

I also get a segmentation fault. But for me it is not clear why?

**Comment by adrpo** [comment]
I will merge this one now.

**Pull request merged**